sktime/utils/tests/test_mlflow_sktime_model_export.py

```diff
@@ -0,0 +1,638 @@
+# -*- coding: utf-8 -*-
+"""Tests for mlflow-sktime custom model flavor."""
+
+__author__ = ["benjaminbluhm"]
+
+import os
+from pathlib import Path
+from unittest import mock
+
+import numpy as np
+import pandas as pd
+import pytest
+import yaml
+
+from sktime.datasets import load_airline, load_longley
+from sktime.datatypes import convert
+from sktime.forecasting.arima import AutoARIMA
+from sktime.forecasting.model_selection import temporal_train_test_split
+from sktime.forecasting.naive import NaiveForecaster
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    """Create a temporary path to save/log model."""
+    return tmp_path.joinpath("model")
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("boto3", "moto", severity="none"),
+    reason="skip test if required soft dependency not available",
+)
+@pytest.fixture(scope="module")
+def mock_s3_bucket():
+    """Create a mock S3 bucket using moto.
+
+    Returns
+    -------
+    string with name of mock S3 bucket
+    """
+    import boto3
+    import moto
```

Comment:
if those 2 dependencies are only required for testing, I would say we can move them to the `dev` dependencies instead `all_extras`.

Reply:
done